### PR TITLE
added filtering capabilities

### PR DESCRIPTION
**Pull request by chrisamiller**
My original plan was to just have this tool add the info, then use GATK VariantFiltration to filter it, but then realized that was more complicated - the filtering is easy to do right here

### CODE REVIEW

File: somatic_llr_filter.py [hunk 1]
```diff
@@ -133,6 +146,17 @@ def create_vcf_writer(args, vcf_reader):
         od = OrderedDict([('ID', args.somatic_field), ('Number', '0'), ('Type', 'Flag'), ('Description', 'Is a somatic mutation')])
         new_header.add_info_line(od)
 
+    #check/add FILTER field in header
+    if not args.llr_threshold == None: #filtering may not even be specified
```
Comment (tmooney) [inline somatic_llr_filter.py]: I believe it's more idiomatic python to write:
```suggestion
    if args.llr_threshold is not None: #filtering may not even be specified
```

**Review by susannasiebert — APPROVED** [review]
+1

File: somatic_llr_filter.py [hunk 2]
```diff
@@ -133,6 +146,17 @@ def create_vcf_writer(args, vcf_reader):
...
+        if args.filter_field in vcf_reader.header.filter_ids():
+            if not args.overwrite:
+                vcf_reader.close()
+                raise Exception("FILTER {} already exists. Choose a different --filter-field, or use the --overwrite flag to retain this filter description and overwrite values".format(args.filter_field))
+        else:
```
Comment (tmooney) [inline somatic_llr_filter.py:155]: What should this do when the field is found and `args.overwrite` is true?

File: Dockerfile [hunk 3]
```diff
@@ -363,10 +363,13 @@ RUN wget --no-check-certificate https://github.com/fulcrumgenomics/fgbio/release
...
-##############
+#######################
+#vcf filtering scripts#
+#######################
 COPY mapq0_vcf_filter.sh /usr/bin/mapq0_vcf_filter.sh
 RUN chmod +x /usr/bin/mapq0_vcf_filter.sh
 RUN pip install pysam==0.11.2.2
 RUN pip install pysamstats
+
+copy somatic_llr_filter.py /usr/bin/somatic_llr_filter.py
```
Comment (susannasiebert) [inline Dockerfile]: I think `COPY` needs to be in all caps.

**Review by susannasiebert — DISMISSED** [review]

File: somatic_llr_filter.py (hunk 2 above)
Comment (chrisamiller) [reply somatic_llr_filter.py:155]: nothing!  The existing filter field definition will be kept

Comment (tmooney) [reply somatic_llr_filter.py:155]: Right! Makes sense...I just confused myself.

**Pull request merged**